### ISSUE TITLE
I keep getting JSONDecodeError

### ISSUE THREAD

**Issue by dekacypher** [nickmccullum/algorithmic-trading-python#23]
My code from the section "Looping Through The Tickers in Our List of Stocks":
<img width="998" alt="Screen Shot 2022-01-19 at 21 15 14" src="https://user-images.githubusercontent.com/87758754/150207206-450afdb4-3342-4b25-8676-4a9287bc43b9.png">

The error:
<img width="1046" alt="Screen Shot 2022-01-19 at 21 15 49" src="https://user-images.githubusercontent.com/87758754/150207225-fc9c7b6b-7896-485e-a890-095b63f0253c.png">

I don't understand what is wrong with my code, i wrote exactly like code from the Algorithmic Trading in Python Course.

And when i print the final_dataframe, the output only shows 484 stocks, and not 505 as in the .csv document? Why is that?

<img width="700" alt="Screen Shot 2022-01-19 at 21 20 05" src="https://user-images.githubusercontent.com/87758754/150207604-f14fae5d-f53b-46fc-9535-df0d62efe622.png">

**Comment by dekacypher** [comment]
**I have found a solution:**
The JSONDecodeError is caused by an HTTP error response that occurs in the midst of the loop, which causes a 404 error and trying to parse this response as JSON is the cause of the JSONDecodeError. Heres where the error occurs:
`data = requests.get(api_url).json()`

**To visualize I printed the status code of each response in the loop:**
<img width="50" alt="Screen Shot 2022-01-20 at 14 45 45" src="https://user-images.githubusercontent.com/87758754/150350499-44264224-2259-4d2c-9811-ce1b05cd69d9.png">

So after researching and trying, i found that by first using if/else to check if the status code of the response is successful (Code: 200) and then continuing to parse the response as JSON is the key to avoid this case.

And now when i print the final_dataframe i get all the 505 sp stocks, because as seen above in the original problem i had, it only printed 484 stocks, this is because the JSONDecodeError gets raised on the 485th stock and therefore STOPS executing the for loop. 
<img width="584" alt="Screen Shot 2022-01-20 at 14 27 12" src="https://user-images.githubusercontent.com/87758754/150351003-6a35d172-1f18-4802-87bd-88e36d172d41.png">

**My solution:**
`final_dataframe = pd.DataFrame(columns = my_columns)
for stock in stocks['Ticker']:
    api_url = f'https://sandbox.iexapis.com/stable/stock/{stock}/quote/?token={IEX_CLOUD_API_TOKEN}'
    data = requests.get(api_url)
    if data.status_code == 200:
        data = data.json()
        final_dataframe = final_dataframe.append(
            pd.Series(
            [
                stock,
                data['latestPrice'],
                data['marketCap'],
                'N/A'
            ],
            index = my_columns),
        ignore_index = True
        )
    else:
        print(f'An error occured, status code: {data.status_code}')
`
